src/compiler.rs

```diff
@@ -758,7 +758,7 @@ fn compile_func(
                                             }
                                         })
                                         .collect::<Result<Vec<_>, _>>()?,
-                                    None => todo!(),
+                                    None => native_panic!("should be Some always"),
```

Comment:
I'm not sure this should always be `Some(_)`, otherwise it'd be an `unreachable!()`, not a `todo!()`. Maybe this error message is more descriptive:
```suggestion
                                    None => native_panic!("not yet implemented"),
```
Yes, it's basically a `todo!()` but without the panic.